core/parse-xml.c

```diff
@@ -1306,6 +1306,8 @@ static void try_to_fill_dive(struct dive *dive, const char *name, char *buf, str
 		return;
 	if (MATCH("visibility.dive", get_rating, &dive->visibility))
 		return;
+	if (MATCH_STATE("airpressure.dive", pressure, &dive->surface_pressure))
```

Comment:
same comment as before about the new flag... this may be correct if we are sure this is initialized to `false`

Reply:
`alloc_dive()` does a `memset()` to 0. I believe that we depend in multiple places on the fact that pointers are zeroed.